frontend/src/__tests__/e2e/CreateEditDeleteProject.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { test, expect } from '@playwright/test';
+
+test.beforeEach(async ({ page }) => {
+  await page.goto('/projects');
+  const projectTestRow = await page.locator('tr', { has: page.locator('text="e2e-test-project"') });
+  const count = await projectTestRow.count();
+  if (count > 0) {
+    await projectTestRow.getByRole('button', { name: 'Actions' }).click();
+    await page.getByRole('menuitem', { name: 'Delete project' }).click();
+    await page.getByRole('textbox', { name: 'Delete modal input' }).fill('e2e-test-project');
+    await page.getByRole('button', { name: 'Delete project' }).click();
+  }
+});
+
+test.afterEach(async ({ page }) => {
+  await page.goto('/projects');
+  const projectTestRow = await page.locator('tr', { has: page.locator('text="e2e-test-project"') });
+  const count = await projectTestRow.count();
+  if (count > 0) {
+    await projectTestRow.getByRole('button', { name: 'Actions' }).click();
+    await page.getByRole('menuitem', { name: 'Delete project' }).click();
+    await page.getByRole('textbox', { name: 'Delete modal input' }).fill('e2e-test-project');
+    await page.getByRole('button', { name: 'Delete project' }).click();
+  }
+});
```

Comment:
Any reason we do this in both BEFORE and AFTER? In case the test breaks midway and we want to cleanup before we start?

Reply:
it can be just for after, but it is just to prevent the case where you had a test not cleanup properly so this catches that case

Reply:
Would you not want to just make the same effort in both spots? It is literal duplication... `await cleanupTestProject(page)` or something...

Reply:
i had there for testing the test purposes. it doesn't need to be in the before

Reply:
i moved it into a helper function and run it before and after now